doc.go

```diff
@@ -0,0 +1,71 @@
+/*
+SQLair is a convenience layer for SQL databases that embeds Go structs directly into SQL queries.
+
+The SQL syntax is expanded with SQLair input and output expressions which indicate parts of the query that correspond to Go structs.
+This allows the user to specify the Go structs they want in the SQL query itself whilst allowing the full power of SQL to be utilised.
+This package also provides an alternative API for reading the rows from the database.
+It does not copy database/sql, it instead improves upon it and removes inconsistencies.
+
+# Basics
+
+To read the results of a query into a struct, the columns listed in the SQL are replaced by a SQLair output expression.
+The output expressions are parsed and replaced with the column names found in the `db` tags of the struct.
+Similarly, to use a query argument directly from a Go struct the question mark is replaced by a SQLair input expression.
+This package does not parse the SQL query, only the SQLair input/output expressions found within it.
```

Comment:
I think we can drop this entire paragraph. The example below says the same thing in a much more clear way.

Reply:
Removed.